types/storybook__addon-knobs/index.d.ts

```diff
@@ -47,13 +48,8 @@ export function color(name: string, value: string, groupId?: string): string;
 
 export function object<T>(name: string, value: T, groupId?: string): T;
 
-export type SelectValue = string | number;
-export function select<T extends string>(name: string, options: { [s: string]: string }, value: T, groupId?: string): T;
-export function select<T extends number>(name: string, options: { [s: number]: string }, value: T, groupId?: string): T;
-export function select<T extends SelectValue>(name: string, options: ReadonlyArray<T>, value: T, groupId?: string): T;
-
-export function selectV2<T extends string | number>(name: string, options: { [s: string]: T | T[] }, value: T | T[], groupId?: string): T;
-export function selectV2<T extends SelectValue>(name: string, options: T[], value: T, groupId?: string): T;
+export type SelectValue = string | number | ReadonlyArray<string> | ReadonlyArray<number>;
+export function select<T extends SelectValue>(name: string, options: { [s: string]: T } | ReadonlyArray<T>, value: T, groupId?: string): T;
```

Comment:
shouldn't the signature for this be what selectV2 was before, given that 4.0 renamed `selectV2` to `select`?

Reply:
That's what I started with, but made additional fixes to the signatures to be in line with the handbook (use `ReadonlyArray<T>` if the function doesn't modify the array), and re-using `SelectValue`. However, going back to the storybook code and double-checking, I see that it doesn't make sense to pass in `(T | T[])[]` for options, so fixing that aspect.